include/nbl/asset/utils/IShaderCompiler.h

```diff
@@ -30,6 +32,7 @@ class NBL_API2 IShaderCompiler : public core::IReferenceCounted
 				{
 					system::path absolutePath = {};
 					std::string contents = {};
+					std::optional<std::array<uint64_t, 4>> hash = std::nullopt; // TODO: we're not yet using IFile::getPrecomputedHash(), so for builtins we can maybe use that in the future
```

Comment:
since you return `contents` and you actually needed to load it in, then the `hash` shouldn't be optional (cause you had to touch the contents anyway)

It would only be the `IFile` that has an optional hash